packages/react/src/Pagination/model.ts

```diff
@@ -135,75 +129,7 @@ export function buildPaginationModel(
     }
   }
 
-  const prev = {type: 'PREV', num: currentPage - 1, disabled: currentPage === 1}
-  const next = {type: 'NEXT', num: currentPage + 1, disabled: currentPage === pageCount}
+  const prev: PageType = {type: 'PREV', num: currentPage - 1, disabled: currentPage === 1}
+  const next: PageType = {type: 'NEXT', num: currentPage + 1, disabled: currentPage === pageCount}
   return [prev, ...pages, next]
 }
-
-export function buildComponentData(
-  page: PageType,
-  hrefBuilder: (n: number) => string,
-  onClick: (e: React.MouseEvent) => void,
-): {
-  props: LinkProps
-  key: string
-  content: string
-} {
-  const props = {}
-  let content = ''
-  let key = ''
-
-  switch (page.type) {
-    case 'PREV': {
-      key = 'page-prev'
-      content = 'Previous'
-      if (page.disabled) {
-        Object.assign(props, {rel: 'prev', as: 'button', 'aria-disabled': 'true'})
-      } else {
-        Object.assign(props, {
-          rel: 'prev',
-          href: hrefBuilder(page.num),
-          'aria-label': 'Previous Page',
-          onClick,
-        })
-      }
-      break
-    }
-    case 'NEXT': {
-      key = 'page-next'
-      content = 'Next'
-      if (page.disabled) {
-        Object.assign(props, {rel: 'next', as: 'button', 'aria-disabled': 'true'})
-      } else {
-        Object.assign(props, {
-          rel: 'next',
-          href: hrefBuilder(page.num),
-          'aria-label': 'Next Page',
-          onClick,
-        })
-      }
-      break
-    }
-    case 'NUM': {
-      key = `page-${page.num}`
-      content = String(page.num)
-      Object.assign(props, {
-        href: hrefBuilder(page.num),
-        // We append "..." to the aria-label for pages that preceed a break because screen readers will
-        // change the tone the text is read in.
-        // This is a slightly nicer experience than skipping a bunch of numbers unexpectedly.
-        'aria-label': `Page ${page.num}${page.precedesBreak ? '...' : ''}`,
-        onClick,
-        'aria-current': page.selected ? 'page' : undefined,
-      })
-      break
-    }
-    case 'BREAK': {
-      key = `page-${page.num}-break`
-      content = '…'
-      Object.assign(props, {as: 'span', role: 'presentation'})
```

Comment:
In the previous implementation, this `as` wasn't working as expected. The element was still rendered as an `<a>` despite the `as` prop here.

Note the rendered HTML:

![image](https://github.com/user-attachments/assets/10ccb336-e427-45e1-b8a0-b6bbcc7bb4fe)

This contributed (along with the tabIndex set [here](https://github.com/primer/brand/pull/1009/files?diff=split&w=0#diff-4242d1a3dcad5607a6d5c3858cbc7b71b628f7962e2ac0d269ecc5d040eaf483L159)) to them still being focusable, and causing https://github.com/github/primer/issues/3734